File: domain/src/main/java/dev/bee/kanjianki/core/StudyRatings.java
package dev.bee.kanjianki.core;

/** Public scheduler rating wire names. */
public final class StudyRatings {
    public static final String AGAIN = "again";
    public static final String HARD = "hard";
    public static final String GOOD = "good";
    public static final String EASY = "easy";

    private StudyRatings() {
    }

    public static String normalize(String rating) {
        if (rating == null) {
            return AGAIN;
        }
        return switch (rating) {
            case AGAIN, HARD, GOOD, EASY -> rating;
            default -> AGAIN;
        };
    }
}

File: domain/src/main/kotlin/dev/bee/kanjianki/core/StudyRatings.kt
package dev.bee.kanjianki.core

/** Public scheduler rating wire names. */
class StudyRatings private constructor() {
    companion object {
        const val AGAIN: String = "again"
        const val HARD: String = "hard"
        const val GOOD: String = "good"
        const val EASY: String = "easy"

        @JvmStatic
        fun normalize(rating: String?): String {
            return when (rating) {
                AGAIN, HARD, GOOD, EASY -> rating
                else -> AGAIN
            }
        }
    }
}
